sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuRegExpReplaceExec.scala

```diff
@@ -31,19 +31,12 @@ class GpuRegExpReplaceMeta(
 
   override def tagExprForGpu(): Unit = {
     expr.regexp match {
-      case Literal(null, _) =>
-        willNotWorkOnGpu(s"null pattern is not supported on GPU")
```

Comment:
I don't see this case handled anymore?  If the null's type isn't UTF8String then it will incorrectly state that a non-literal pattern is not supported (but it is a literal).  If the null's type is UTF8String then it looks like we'll NPE when we try to convert it to a string below.

Reply:
It shouldn't be possible to have a null literal in this context since spark would translate it to a `cast(null as string)` but I will make this more defensive and update the error messages.